### DISCUSSION TITLE
`IndeiValidationPlugin` not work

### DISCUSSION THREAD

**Discussion by corrm** [AvaloniaUI/Avalonia#16818] [Q&A]
My `ViewModelBase`:
```CSharp
using CommunityToolkit.Mvvm.ComponentModel;

public abstract class ViewModelBase : ObservableValidator
{
}
```

In `OnFrameworkInitializationCompleted` i add:
```CSharp
BindingPlugins.DataValidators.RemoveAt(0);
```

MyViewModel:
```CSharp
public sealed partial class InitializePageViewModel : ViewModelBase
{
    [ObservableProperty]
    [Required]
    private string _processId;
}
```

MyView:
```XAML
<TextBox
    Grid.Column="0"
    Watermark="0x15AB OR 1234"
    Text="{Binding ProcessId}" />
```
The result:
![image](https://github.com/user-attachments/assets/0e3e1ca8-d1e2-41ee-8452-d7bb2a5116bf)

Only if i remove `BindingPlugins.DataValidators.RemoveAt(0);` and `DataAnnotationsValidationPlugin` take control:
![image](https://github.com/user-attachments/assets/7afaf8ec-4290-4321-ba69-026c20903b15)

I should get double validation but i get one.

So why `IndeiValidationPlugin` not work? and i use `CommunityToolkit.Mvvm`

**Comment by stevemonaco** [comment]
https://github.com/AvaloniaUI/Avalonia/blob/06ece8da8813119b0feb0d35de397dd031157095/src/Avalonia.Base/Data/Core/Plugins/BindingPlugins.cs#L18-L23

The 0-index (which you remove) is currently the `DataAnnotationsValidationPlugin`, so the `IndeiValidationPlugin` is what is providing the validation error. Unless I'm misunderstanding something here.

**Reply by stevemonaco** [reply]
You may need `[NotifyDataErrorInfo]` on your field like https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#requesting-property-validation describes.

**Reply by corrm** [reply]
> https://github.com/AvaloniaUI/Avalonia/blob/06ece8da8813119b0feb0d35de397dd031157095/src/Avalonia.Base/Data/Core/Plugins/BindingPlugins.cs#L18-L23
> 
> The 0-index (which you remove) is currently the `DataAnnotationsValidationPlugin`, so the `IndeiValidationPlugin` is what is providing the validation error. Unless I'm misunderstanding something here.

There problem is there is no validation error at all.

**Reply by corrm** [reply]
> You may need `[NotifyDataErrorInfo]` on your field like https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#requesting-property-validation describes.

Mostly that's how it works. `[NotifyDataErrorInfo]` is needed. will keep this discussion open if there is another solution.
